pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -503,6 +503,14 @@ void connectionOpened(final ClientCnx cnx) {
                 cnx.channel().close();
                 return null;
             }
+            
+            if (client.getConfiguration().getConnectionsPerBroker() == 0) {
```

Comment:
You are right - the fix won't work in these scenarios .. I will try another approach - call close in the finalize method of ClientCnx